### ISSUE TITLE
UnnecessaryLet but let is necessary

### ISSUE THREAD

**Issue by MassimoDjungle** [detekt/detekt#7732]
This is the affected snippet:
```
firebaseAuth.currentUser?.takeIf { it.isAnonymous }?.let { currentAnonymousUser ->
    currentAnonymousUser.linkWithCredential(phoneAuthCredential)
    ...
}
```

## Expected Behavior
The let is not "unnecessary", the error should not happen.

## Your Environment
* Version of detekt used: 1.23.7
* Version of Gradle used (if applicable): 8.9
* Operating System and version: MacOS X Sonoma 14.6.1

**Comment by BraisGabin** [comment]
What's the error message from detekt? I think that this rule advise about what change should be done. To my taste the `let` only makes the code more difficult to read and a good old `if` does the job perfectly. 

```kotlin
val currentAnonymousUser = firebaseAuth.currentUser?.takeIf { it.isAnonymous }
if (currentAnonymousUser != null) {
    currentAnonymousUser.linkWithCredential(phoneAuthCredential)
    // ...
}
```

or even better:

```kotlin
val currentAnonymousUser = firebaseAuth.currentUser
if (currentAnonymousUser != null && currentAnonymousUser.isAnonymous) {
    currentAnonymousUser.linkWithCredential(phoneAuthCredential)
    // ...
}
```

or:

```kotlin
val currentAnonymousUser = firebaseAuth.currentUser
if (currentAnonymousUser?.isAnonymous == true) {
    currentAnonymousUser.linkWithCredential(phoneAuthCredential)
    // ...
}
```

And if the compiler can ensure that `firebaseAuth.currentUser` always returns the same:

```kotlin
if (firebaseAuth.currentUser?.isAnonymous == true) {
    firebaseAuth.currentUser.linkWithCredential(phoneAuthCredential)
    // ...
}
```

**Comment by MassimoDjungle** [comment]
you're right, `currentAnonymousUser` was used just once; it wasn't adviced in detekt 1.23.6
